datacite-transform/build.gradle

```diff
@@ -1,41 +1,80 @@
-import java.nio.file.Files
-import java.nio.file.Paths
-
 plugins {
     id 'java'
-    id "org.unbroken-dome.xjc" version "2.0.0"
+    id("io.github.alexpercont.xjc-plugin") version "1.1"
+    id "de.undercouch.download" version "5.6.0"
 }
 
 group 'no.unit.nva'
 
 dependencies {
-    implementation group: 'javax.xml.bind', name: 'jaxb-api', version: '2.3.1'
-    implementation group: 'jakarta.xml.bind', name: 'jakarta.xml.bind-api', version: '2.3.2'
-    runtimeOnly group: 'org.glassfish.jaxb', name: 'jaxb-runtime', version: '2.3.1'
+    implementation group: 'jakarta.xml.bind', name: 'jakarta.xml.bind-api', version: '4.0.2'
+    runtimeOnly group: 'org.glassfish.jaxb', name: 'jaxb-runtime', version: '4.0.5'
 }
 
-def schemaDir = "${buildDir}/tmp/schema/"
+ext.SCHEMA_VERSION = '4.5'
+ext.SCHEMA_DIR = file(layout.buildDirectory.asFile.get().toURI().toString() + '/tmp/schema')
+ext.METADATA_XSD = 'metadata.xsd'
+ext.DID_DOWNLOAD = new HashMap<String, Boolean>()
+
+tasks.register('fetchSchema') {
+    doLast {
+        getUris().forEach {
+            downloadFile(it.toString())
+        }
+        if (DID_DOWNLOAD.any { it.value == false }) {
+            System.out.println("WARN: Some XSD files were cached from a previous run, to avoid this, run ./gradlew clean")
+        }
+    }
+}
 
-task fetchSchema() {
-    def schemaVersion = '4.4'
+private List<String> getUris() {
 
-    mkdir(schemaDir)
+    def source = downloadFile(METADATA_XSD)
 
-    def urlFile = "datacite.url"
+    def xml = new groovy.xml.XmlSlurper().parseText(source.getText())
 
-    if (!file(schemaDir + urlFile).exists()) {
-        Files.createFile(Paths.get(schemaDir + urlFile))
-                .text = "https://schema.datacite.org/meta/kernel-${schemaVersion}/metadata.xsd"
+    return xml.'**'.findAll {
+        it.name() == 'include' || it.name() == 'import'
+    }.collect {
+        it.@'schemaLocation'
     }
```

Comment:
Here, we're extracting the URIs of the XML files that we need to gather in order to be able to build.